qiskit_nature/transformers/particle_hole_transformer.py

```diff
@@ -28,5 +45,2102 @@ def transform(self, q_molecule: QMolecule) -> QMolecule:
+        self._h1_transform(unitary_matrix)
+        self._h2_transform(unitary_matrix)
+
+    def _h1_transform(self, unitary_matrix):
+        """
+        Transform h1 based on unitary matrix, and overwrite original property.
+        Args:
+            unitary_matrix (numpy.ndarray): A 2-D unitary matrix for h1 transformation.
+        """
+        self._h1 = unitary_matrix.T.conj().dot(self._h1.dot(unitary_matrix))
+
+    def _h2_transform(self, unitary_matrix):
+        """
+        Transform h2 to get fermionic hamiltonian, and overwrite original property.
+        Args:
+            unitary_matrix (numpy.ndarray): A 2-D unitary matrix for h1 transformation.
+        """
+        num_modes = unitary_matrix.shape[0]
+        temp_ret = np.zeros((num_modes, num_modes, num_modes, num_modes),
+                            dtype=unitary_matrix.dtype)
+        unitary_matrix_dagger = np.conjugate(unitary_matrix)
+
+        # option 3: temp1 is a 3-D tensor, temp2 and temp3 are 2-D tensors
+        # pylint: disable=unsubscriptable-object
+        for a_i in range(num_modes):
+            temp1 = np.einsum('i,i...->...', unitary_matrix_dagger[:, a_i], self._h2)
+            for b_i in range(num_modes):
+                temp2 = np.einsum('j,j...->...', unitary_matrix[:, b_i], temp1)
+                temp3 = np.einsum('kc,k...->...c', unitary_matrix_dagger, temp2)
+                temp_ret[a_i, b_i, :, :] = np.einsum('ld,l...c->...cd', unitary_matrix, temp3)
+
+        self._h2 = temp_ret
+
+    def _sort(self, seq):
+        """
+        Tool function for normal order, should not be used separately
+
+        Args:
+            seq (list): array
+
+        Returns:
+            list: integer e.g. swapped array, number of swaps
+        """
+        swap_counter = 0
+        changed = True
+        while changed:
+            changed = False
+            for i in range(len(seq) - 1):
+                if seq[i] > seq[i + 1]:
+                    swap_counter += 1
+                    seq[i], seq[i + 1] = seq[i + 1], seq[i]
+                    changed = True
+
+        return seq, swap_counter
+
+    def last_two_indices_swap(self, array_ind_two_body_term):
+        """
+        Swap 2 last indices of an array
+
+        Args:
+            array_ind_two_body_term (list): TBD
+
+        Returns:
+            list: TBD
+        """
+        swapped_indices = [0, 0, 0, 0]
+        swapped_indices[0] = array_ind_two_body_term[0]
+        swapped_indices[1] = array_ind_two_body_term[1]
+        swapped_indices[2] = array_ind_two_body_term[3]
+        swapped_indices[3] = array_ind_two_body_term[2]
+
+        return swapped_indices
+
+    def _normal_order_integrals(self, n_qubits, n_occupied, array_to_normal_order,
+                               array_mapping, h1_old, h2_old,
+                               h1_new, h2_new):
+        """
+        Given an operator and the rFs and rsgtu from Gaussian it produces new
```

Comment:
Can we amend this comment - it has been in the old code forever and never fixed. The logic may have been developed originally under that driver but now the qmolecule/integrals are driver independent